### PR TITLE
System installation

### PR DESCRIPTION
**Pull request by kavics**
"First installation" problem fixed and testable but 2 unit tests are failed.

### CODE REVIEW

File: src/ContentRepository/Packaging/Steps/Internal/CheckDatabaseConnection.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	DROP TABLE [dbo].[temptable]
+END
+";
+        #endregion
+
+        public override void Execute(ExecutionContext context)
+        {
+            var attempt = 0;
+            Exception lastError = null;
+            while (++attempt < 2)
```
Comment (tusmester) [inline src/ContentRepository/Packaging/Steps/Internal/CheckDatabaseConnection.cs]: Shouldn't this be bigger than 2?

Comment (kavics) [inline src/ContentRepository/Packaging/Steps/Internal/CheckDatabaseConnection.cs]: Sorry, fixed. It was a value for demo.

**Comment by kavics** [comment]
There is a problem with the system install (system install = running the package: 'Install' the 'SenseNet.Services'). Let's see the code:

``` csharp
if (PackageType == PackageType.Install)
{
    if (!SystemInstall && existingComponentInfo != null)
        throw new PackagePreconditionException(string.Format(SR.Errors.Precondition.CannotInstallExistingComponent1, this.ComponentId),
            PackagingExceptionType.CannotInstallExistingComponent);
}
```

It saids: everythig was protected against the overwriting except the most basic comonent and the database. It was not possible to reinstall anything but the whole database could dropped instantly without confirmation. I think it was a little bit dangerous for anyone who could go close to the web\Admin directory... But deleting 18 character can protect the production database:

``` csharp
if (PackageType == PackageType.Install)
{
    if (existingComponentInfo != null)
        throw new PackagePreconditionException(string.Format(SR.Errors.Precondition.CannotInstallExistingComponent1, this.ComponentId),
            PackagingExceptionType.CannotInstallExistingComponent);
}
```
After that there is 2 problems:
- Developer's recurring workflow does not work: pull / compile / reset database. (quick solution: developers may make 'Tool' from 'Install')
- Cannot install the main component if the desired database exists.

So what is the good solution?

**Comment by kavics** [comment]
Done (documentation is in progress).

**Comment by kavics** [comment]
Very done.
For the common mindset: only the system component (SenseNet.Services) can be reinstalled. The general forced reinstalling operation was not scope. The system install drops the database so the second install is a clean install too. Before reinstalling another component need to do something with old packaging records. We don't like this game now.

**Pull request merged**